cpu/atmega_common/periph/i2c.c

```diff
@@ -120,169 +133,95 @@ int i2c_init_master(i2c_t dev, i2c_speed_t speed)
     /* enable device */
     TWCR |= (1 << TWEN);
 
-    return 0;
+    return;
 }
 
-int i2c_acquire(i2c_t dev)
+int i2c_read_bytes(i2c_t dev, uint16_t addr, void *data, size_t len, uint8_t flags)
 {
-    if (!(dev < I2C_NUMOF)) {
-        return -1;
-    }
-    mutex_lock(&lock);
-    return 0;
-}
+    assert (len > 0);
 
-int i2c_release(i2c_t dev)
-{
     if (!(dev < I2C_NUMOF)) {
-        return -1;
-    }
-    mutex_unlock(&lock);
-    return 0;
-}
-
-int i2c_read_byte(i2c_t dev, uint8_t address, void *data)
-{
-    return i2c_read_bytes(dev, address, data, 1);
-}
-
-int i2c_read_bytes(i2c_t dev, uint8_t address, void *data, int length)
-{
-    assert(length > 0);
-
-    if (!(dev < I2C_NUMOF)) {
-        return -1;
+        return I2C_ERR;
     }
 
     uint8_t *my_data = data;
+
     /* send start condition and slave address */
-    if (_start(address, I2C_FLAG_READ) != 0) {
-        return 0;
+    if(~(flags & I2C_NOSTART)) {
+        if (_start(addr, ATMEGA_I2C_FLAG_READ) != 0) {
+            return I2C_ADDR_NACK;
+        }
     }
 
-    for (int i = 0; i < length; i++) {
-        /* Send NACK for last received byte */
-        if ((length - i) == 1) {
-            TWCR = (1 << TWEN) | (1 << TWINT);
-        }
-        else {
-            TWCR = (1 << TWEA) | (1 << TWEN) | (1 << TWINT);
-        }
-        DEBUG("Wait for byte %i\n", i+1);
+    for (size_t i = 0; i < len; i++) {
+        TWCR = (1 << TWEN) | (1 << TWINT);
+        DEBUG("Wait for byte\n");
+
         /* Wait for TWINT Flag set. This indicates that DATA has been received.*/
         while (!(TWCR & (1 << TWINT))) {}
+
         /* receive data byte */
-        my_data[i] = TWDR;
-        DEBUG("Byte %i received\n", i+1);
+        my_data[0] = TWDR;
+        DEBUG("Byte received\n");
     }
 
     /* end transmission */
-    _stop();
-
-    return length;
-}
-
-int i2c_read_reg(i2c_t dev, uint8_t address, uint8_t reg, void *data)
-{
-    return i2c_read_regs(dev, address, reg, data, 1);
-}
-
-int i2c_read_regs(i2c_t dev, uint8_t address, uint8_t reg, void *data, int length)
-{
-    assert(length > 0);
-
-    if (!(dev < I2C_NUMOF)) {
-        return -1;
-    }
-
-    /* start transmission and send slave address */
-    if (_start(address, I2C_FLAG_WRITE) != 0) {
-        return 0;
-    }
-
-    /* send register address and wait for complete transfer to be finished*/
-    if (_write(&reg, 1) != 1) {
+    if(~(flags & I2C_NOSTOP))
         _stop();
-        return 0;
-    }
-
-    /* now start a new start condition and receive data */
-    return i2c_read_bytes(dev, address, data, length);
-}
 
-int i2c_write_byte(i2c_t dev, uint8_t address, uint8_t data)
-{
-    return i2c_write_bytes(dev, address, &data, 1);
+    return I2C_ACK;
```

Comment:
I2C_ACK doesn't exist anymore as we switch to errno return code for the whole API. You can replace I2C_ACK by 0